SPECS/dracut/20overlayfs/overlayfs-mount.sh

```diff
@@ -54,49 +51,51 @@ parse_cmdline_args() {
 
     # Retrieve the OverlayFS parameters.
     [ -z "${overlayfs}" ] && overlayfs=$(getarg rd.overlayfs=)
-    # Retrieve the persistent volume for the OverlayFS.
-    [ -z "${overlayfs_persistent_volume}" ] && overlayfs_persistent_volume=$(getarg rd.overlayfs_persistent_volume=)
 }
 
-# Modified function to mount the physical partition
-mount_physical_partition() {
-    mkdir -p "${OVERLAY_MOUNT}"
-    # Leverage the partition from cmdline
-    local partition="${overlayfs_persistent_volume}"
+# Modified function to mount volatile or persistent volume.
+mount_volatile_persistent_volume() {
+    local _volume=$1
+    local _overlay_mount=$2
 
-    if [ -z "${partition}" ]; then
-        # Fallback to volatile overlay if no persistent volume is specified
+    mkdir -p "${_overlay_mount}"
+
+    if [[ "${_volume}" == "volatile" ]]; then
+        # Fallback to volatile overlay if no persistent volume is specified.
         echo "No overlayfs persistent volume specified. Creating a volatile overlay."
-        mount -t tmpfs tmpfs -o ${OVERLAY_MNT_OPTS} "${OVERLAY_MOUNT}" || \
-            die "Failed to create overlay tmpfs at ${OVERLAY_MOUNT}"
+        mount -t tmpfs tmpfs -o ${OVERLAY_MNT_OPTS} "${_overlay_mount}" || \
+            die "Failed to create overlay tmpfs at ${_overlay_mount}"
     else
         # Check if /etc/mdadm.conf exists.
         if [ -f "/etc/mdadm.conf" ]; then
-            mdadm --assemble ${partition} || \
+            mdadm --assemble ${_volume} || \
```

Comment:
Not needed to fix here, but dracut should already be able to assemble mdadm raid on its own. If you add an entry to the init queue to wait for it I think that should work?

Reply:
Mark here, thanks.